### DISCUSSION TITLE
How ESXi network configuration was modeled in NetBox

### DISCUSSION THREAD

**Discussion by daniel-zook** [netbox-community/netbox#13401] [General]
I recently started documenting our ESXi servers using NetBox, including the networking configuration.  We have two physical NICs connected to a VMWare Virtual Switch, and the VMkernel adapters and VMs connect to the Virtual Switch.  The main purpose here is to document the configuration of the ESXi host.  (The network configuration of the VMs themselves is not in-scope in this case.)

I saw a few posts about how one might do this, so I'm sharing this here in the hopes that it might be useful to someone.

Attached is a screen capture showing the relevant information.  If anyone has any questions about this, please reply to this thread and I'll do my best to explain.

![Example documenting ESXi networking config](https://github.com/netbox-community/netbox/assets/102061371/b9d192ff-e281-4c40-a285-5ab44631f51a)

**Comment by zmaster7** [comment]
Thanks for sharing, very similar to the way I have mine setup! One thing you may want to include is the physical port labels for your vmnics, if you haven't done so already. 

In your case it's not a big deal since you have both SFP+ connections on the same vSwitch, but it does make things easier once you start getting into multiple vSwitches/physical NICs.

**Reply by daniel-zook** [reply]
Good suggestion.  (Unfortunately, in our case the server doesn't have physical labels on the port other than "1", "2", etc. for the on-board interfaces.  I'll add those at least.)

**Comment by FRACKPRINTERS** [comment]
This is really great!
Have you figured out how to model Port Groups or Distributed Switches?

**Comment by zmaster7** [comment]
> This is really great! Have you figured out how to model Port Groups or Distributed Switches?

If you really wanted to model port groups, you could probably add them as additional virtual interfaces much like the vmk ports above and add the untagged VLAN if necessary. In my case, we label our port groups "VLAN ####," so having them defined on the physical interfaces is enough for us.

As far as distributed switches, I would assume they would be modeled the same way in NetBox as standard vSwitches, or I suppose you could add all the hosts as a virtual chassis and manage one set of interfaces tied between all of them that way. We try to keep our environment as simple as possible with our "small" deployment of 300-400 VMs, so we are still on standard vSwitches.

**Comment by Blue-Goggins** [comment]
Have you uploaded this as a template in the library?

**Reply by is-info** [reply]
No, I have not uploaded a template for this.  I don't know how to do that.  If you think it would be worthwhile, I'm open to it.  I assume you are referring to this?  https://github.com/netbox-community/devicetype-library/blob/master/CONTRIBUTING.md
